lib/possible_bipartition.rb

```diff
@@ -1,4 +1,44 @@
+# Time: O(n + e) - work with each node and with each edge
+# Space: O(n + e) - created 2 additional arrays + use of system stack for recursion
+# from 'Algorithms' by Robert Sedgewick
+class Graph
```

Comment:
👍  Interesting that you added a class.  Good recursive DFS solution.  Well done.

Reply:
Thank you, Chris! I added a class there so that I could have shared variables instead of passing them all to my dfs function.